src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -350,6 +354,363 @@ def output_gds(self, value: str) -> None:
+        pass
+
+    def get_additional_drc_text(self) -> str:
+        """ Get the additional custom DRC command text to add after the boilerplate commands at the top of the DRC run file. """
+
+        # Mode can be auto, manual, append, or prepend
+        add_drc_text_mode = str(self.get_setting("drc.inputs.additional_drc_text_mode"))
+
+        # manul_add_drc_text will only be used in manual, append, and prepend modes
+        manual_add_drc_text = str(self.get_setting("drc.inputs.additional_drc_text"))
+
+        # tech_add_drc_text will only be used in auto, append, and prepend modes
+        tech_add_drc_text = get_or_else(self.technology.additional_drc_text, "") # type: str
+
+        # Default to auto (use tech_add_drc_text)
+        add_drc_text = tech_add_drc_text
+
+        if add_drc_text_mode == "auto":
+            pass
+        elif add_drc_text_mode == "manual":
+            add_drc_text = manual_add_drc_text
+        elif add_drc_text_mode == "append":
+            add_drc_text = tech_add_drc_text + manual_add_drc_text
+        elif add_drc_text_mode == "prepend":
+            add_drc_text = manual_add_drc_text + tech_add_drc_text
+        else:
+            self.logger.error(
+                "Invalid additional_drc_text_mode {mode}. Using auto.".format(mode=add_drc_text_mode))
+
+        return add_drc_text
+
+    @property
+    def layout_file(self) -> str:
+        """
+        Get the input layout file (e.g. *.gds)
+
+        :return: The input layout file (e.g. *.gds)
+        """
+        try:
+            return self.attr_getter("_layout_file", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the input layout file (e.g. *.gds) yet")
+
+    @layout_file.setter
+    def layout_file(self, value: str) -> None:
+        """ Set the input layout file (e.g. *.gds)"""
+        if not (isinstance(value, str)):
+            raise TypeError("layout_file must be a str")
+        self.attr_setter("_layout_file", value)
+
+    ### Outputs ###
+
+    @abstractmethod
+    def drc_results_pre_waived(self) -> Dict[str, int]:
+        """ Return a Dict mapping the DRC check name to an error count (pre-waivers). """
+        pass
+
+    def signoff_results(self) -> int:
+        """ Return the count of unwaived DRC errors. """
+        return reduce(lambda a, b: a + b, self.drc_results().values())
+
+    def drc_results(self) -> Dict[str, int]:
+        """ Return a Dict mapping the DRC check name to an error count (with waivers). """
+        return {k: 0 if k in self.waived_drc_rules() else int(v) for k, v in self.drc_results_pre_waived()}
+
+
+class HammerLVSTool(HammerSignoffTool):
+    @abstractmethod
+    def fill_outputs(self) -> bool:
+        pass
+
+    ### Inputs ###
+
+    @abstractmethod
+    def waived_erc_rules(self) -> List[str]:
+        # TODO [John] how to waive specific instances of ERC rules, rather than blanket waivers
+        # TODO [John] should this go in the YAML file?
```

Comment:
What do these waivers look like? If they're pretty simple it might be worth it to put into the config.

Reply:
For blanket waivers, it's just a list of strings. IMO config is a fine place for it. We will need to implement instance- or area-specific waivers at a future time, since that is a lot more complicated, and generally not a feature we use much.

Reply:
Maybe we could call this `blanket_waived_erc_rules` or something?